deepaas/cmd/execute.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python -*- coding: utf-8 -*-
+
+# Copyright 2020 Spanish National Research Council (CSIC)
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import magic
+import mimetypes
+import os
+import shutil
+import sys
+
+from oslo_config import cfg
+from oslo_log import log
+
+from deepaas.model import loading
+from deepaas.model.v2.wrapper import UploadedFile
+
+cli_opts = [
+    cfg.StrOpt('input_file',
+               short="i",
+               help="""
+Set local input file to predict.
+"""),
+    cfg.StrOpt('content_type',
+               default='application/json',
+               short='ct',
+               help="""
+Especify the content type of the output file. The selected
+option must be available in the model used.
+(by default application/json).
+"""),
+    cfg.StrOpt('output',
+               short="o",
+               help="""
+Save the result to a local file.
+"""),
+    cfg.BoolOpt('url',
+                short='u',
+                default=False,
+                help="""
+Run as input file an URL.
+If this option is set to True, we can use the URL
+of an image as an input file.
+"""),
+]
+
+CONF = cfg.CONF
+CONF.register_cli_opts(cli_opts)
+
+LOG = log.getLogger(__name__)
+
+# Loading the model installed
+
+
+def model_name():
+    global MODEL_NAME
+    try:
+        for name, model in loading.get_available_models("v2").items():
+            MODEL_NAME = name
```

Comment:
Here you are iterating over a list of models, keeping the last one as the model to be used. However, several models can be present in the system.

I would suggest to use an option `--model-name`. If the option is set, we would look for that model name. If it is not set and only one model is present, we use that one. If it is not set, and several models are available, we fail.